neptune-core/src/state/wallet/wallet_file.rs

```diff
@@ -2,28 +2,133 @@ use std::fs;
 use std::path::Path;
 use std::path::PathBuf;
 
-use anyhow::ensure;
 use anyhow::Context;
 use anyhow::Result;
+use anyhow::{anyhow, ensure};
 use rand::rng;
 use rand::Rng;
 use serde::Deserialize;
 use serde::Serialize;
-use tracing::info;
-use zeroize::ZeroizeOnDrop;
+use tracing::{error, info, warn};
+use zeroize::{Zeroize, ZeroizeOnDrop, Zeroizing};
 
+use super::encryption::{EncryptedWalletFile, PasswordManager};
 use super::secret_key_material::SecretKeyMaterial;
 use super::wallet_entropy::WalletEntropy;
 
 pub const WALLET_DIRECTORY: &str = "wallet";
 pub const WALLET_SECRET_FILE_NAME: &str = "wallet.dat";
+pub const WALLET_ENCRYPTED_FILE_NAME: &str = "wallet.encrypted";
+pub const WALLET_BACKUP_SUFFIX: &str = ".backup";
 pub const WALLET_OUTGOING_SECRETS_FILE_NAME: &str = "outgoing_randomness.dat";
 pub const WALLET_INCOMING_SECRETS_FILE_NAME: &str = "incoming_randomness.dat";
 const STANDARD_WALLET_NAME: &str = "standard_wallet";
 const STANDARD_WALLET_VERSION: u8 = 0;
 pub const WALLET_DB_NAME: &str = "wallet";
 pub const WALLET_OUTPUT_COUNT_DB_NAME: &str = "wallout_output_count_db";
 
+/// Environment variable for wallet password (INSECURE - only for testing/automation)
+pub const WALLET_PASSWORD_ENV_VAR: &str = "NEPTUNE_WALLET_PASSWORD";
+
+/// Password source for wallet encryption
+#[derive(Debug, Clone)]
+pub enum PasswordSource {
+    /// Interactive prompt (most secure)
+    Interactive,
+    /// From CLI argument (insecure, for testing only)
+    Cli(String),
+    /// From environment variable (insecure, for automation)
+    Environment(String),
```

Comment:
Why is the environment variable considered insecure? Only the user owning the process can see the env variable. Cf.
https://security.stackexchange.com/a/14009/168615